src/figuresClasses.ts

```diff
@@ -1,11 +1,94 @@
-export interface Figure {}
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
+
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error(
+        'The length of each side can`t be negative or equal zero',
+      );
+    }
+
+    if (
+      this.a + this.b <= this.c ||
+      this.a + this.c <= this.b ||
+      this.b + this.c <= this.a
+    ) {
+      throw new Error(
+        'The sum of any two sides must be greater than the remaining side.',
+      );
+    }
+  }
+
+  getArea(): number {
+    let area = 0;
+    const p = (this.a + this.b + this.c) / 2;
 
-export class Triangle implements Figure {}
+    area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
 
-export class Circle implements Figure {}
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape = Shape.circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('The radius can`t be negative or equal zero');
```

Comment:
The error message could be more descriptive by including information about the context of the error, such as 'Circle radius must be positive'.